src/main/java/com/rackspace/jenkins_nodepool/NodePoolJob.java

```diff
@@ -96,15 +175,57 @@ void setFinishTime() {
         }
 
         long getDurationSeconds() {
-            return finishTime - startTime;
+            long end = finishTime;
+            if (end == 0) {
+                end = System.currentTimeMillis();
+            }
+
+            final long millis = end - startTime;
+            return millis / 1000L;
+        }
+
+        public String getError() {
+            if (e == null) {
+                return null;
+            } else {
+                return ExceptionUtils.getStackTrace(e);
+            }
         }
 
         public boolean isDone() {
             return finishTime != 0L;
         }
 
+        public boolean isFailure() {
+            return isDone() && e != null;
```

Comment:
Would think that just `e != null` alone would indicate a failure (regardless of done/finish time value).

Reply:
That's true